update dependencies to support latest Gren version(0.4.4)

ajstrand opened the pull request:
To support the latest version of Gren(0.4.4), the dependencies in this package were updated. 

Let me know ifd there's anything else that would need to be done.

ajstrand committed:
update dependencies to support latest gren version

joeybright committed:
Update `mimeTypes` given there's no `Dict.fromArray` anymore

joeybright commented:
@ajstrand Thanks for reminding me this needs to be done! There are a few errors that I noticed when building the package - I had enough time tonight to fix up the `mimeTypes` issues. There's some refactoring that needs to be done given the file system part of the node package changed quite substantially in the latest version. I'll try to get to it later this week, but would welcome any progress towards the required changes!

ajstrand commented:
@joeybright thanks for the comment! Im new to gren/elm. 

I can take a look at file system API refactoring and see if im able to help at all there. 

Is building the package with gren/node enough to spit out errors that i can work on fixing? 

Or are there more steps to test the package?

joeybright commented:
@ajstrand You should be able to clone the package locally, run `gren make` in the project root, and it'll spit out any errors. I don't have a test suite for this package (the built-in gren testing doesn't support testing `Task` type for good reasons, which this package mainly uses and returns), but I do have some private codebases that leverage the package. So, when `gren make` runs without error, I can test the updates on those!

joeybright committed:
Refactored to work with 0.4.4 and the new `FileSystem.Path.Path` type

The package is compiling with Gren 0.4.4. compiler. But nothing's been tested yet, so the implementation might not be 100% correct.

joeybright commented:
@ajstrand I got some time and was able to get rewrite the package to get it compiling with 0.4.4! I have yet to do any testing, though, and won't be able to get to it for a bit. So, if you're still interested in contributing, writing a test app would be a great help. Otherwise, I'll do some testing when I have the time. If you need any help or have any questions, let me know!

joeybright committed:
Fixed problem with getting the wrong MIME type returned

With the new `Path` type in the `FileSystem` module, there's no need to check for a `"."`.

joeybright committed:
Considering the directory in the config when looking up default `index.html` file

blaix commented on src/Server/Static.gren:
```diff
@@ -90,91 +73,31 @@ type alias Config =
-    -- Build the path from the given path and any modifiers. Leverages the `FileSystem`
-    -- built-ins to normalize it.
-    path
-        |> Array.pushFirst directory
```
Testing this branch in prettynice and getting errors like:

```
Error "ENOENT" "ENOENT: no such file or directory, open '/main.js'
```

I'm guessing because `directory` is no longer being applied?

blaix replied:
This seems to fix it: https://github.com/blaix/gren-static-serve/commit/85bdd2a79232a4c75a5eeacd373695c29803067a

But I haven't tested SPA mode.

joeybright commented:
@blaix Thanks for looking into this a bit more! I spent some time earlier this morning testing. I rewrote a large portion of the logic so that everything works as expected. I'd like to do a bit more testing tonight before committing the code, though. If you have the bandwidth, after I commit my updates, it'd be helpful if you could make sure they work for you. Having another pair of eyes is always good!

joeybright committed:
Fixing bugs after testing each mode

blaix commented:
@joeybright The changes are working great for me, thanks!
